service/history/api/startworkflow/api.go

```diff
@@ -431,6 +438,35 @@ func (s *Starter) resolveDuplicateWorkflowID(
 		return nil, StartNew, nil
 	}
 
+	killOnce.Do(func() {
+		err = api.GetAndUpdateWorkflowWithNew(
+			ctx,
+			nil,
+			definition.NewWorkflowKey(
+				s.namespace.ID().String(),
+				workflowID,
+				currentWorkflowConditionFailed.RunID,
+			),
+			func(workflowLease api.WorkflowLease) (*api.UpdateWorkflowAction, error) {
+				mutableState := workflowLease.GetMutableState()
+				return api.UpdateWorkflowTerminate, workflow.TerminateWorkflow(
+					mutableState,
+					"TESTING",
+					nil,
+					"TESTING",
+					false,
+					nil,
+				)
+			},
+			nil,
+			s.shardContext,
+			s.workflowConsistencyChecker,
+		)
+		if err != nil {
+			panic(err)
+		}
+	})
+
```

Comment:
Obviously, I had to remove this again, but it demonstrates that the change works.